3-java-servlet-web-app/1-Authentication/sign-in-ciam/src/main/java/com/microsoft/azuresamples/msal4j/authservlets/AADRedirectServlet.java

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azuresamples.msal4j.authservlets;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.microsoft.azuresamples.msal4j.helpers.AuthException;
+import com.microsoft.azuresamples.msal4j.helpers.AuthHelper;
+import com.microsoft.azuresamples.msal4j.helpers.Config;
+import com.microsoft.azuresamples.msal4j.helpers.IdentityContextAdapterServlet;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class defines the endpoint for processing the redirect from AAD MSAL
+ * Java apps using this sample's paradigm will require this.
+ */
+@WebServlet(name = "AADRedirectServlet", urlPatterns = "/auth/redirect")
+public class AADRedirectServlet extends HttpServlet {
+    private static Logger logger = Logger.getLogger(AADRedirectServlet.class.getName());
+
+    @Override
+    protected void doGet(final HttpServletRequest req, final HttpServletResponse resp)
+            throws ServletException, IOException {
+        logger.log(Level.FINE, "Request has come with params {0}", req.getQueryString());
+        try {
+            AuthHelper.processAADCallback(new IdentityContextAdapterServlet(req, resp));
+            logger.log(Level.INFO, "redirecting to home page.");
+            resp.sendRedirect(Config.HOME_PAGE);
+        } catch (AuthException ex) {
+            logger.log(Level.WARNING, ex.getMessage());
```

Comment:
nit: use ERROR level ?
nit: isn't there an overload of `log` that deals with exceptions?